### ISSUE TITLE
Release public Ip of System VMs should FAIL

### ISSUE THREAD

**Issue by weizhouapache** [apache/cloudstack#9275]
Try to release the public IP of system vm

Prior

![image](https://github.com/apache/cloudstack/assets/57355700/a98f8be6-7cff-445e-b509-d1bf7d5d2624)

remove

![image](https://github.com/apache/cloudstack/assets/57355700/274b70e1-e866-4cbc-9fb8-c35fca0af169)

After

![image](https://github.com/apache/cloudstack/assets/57355700/513e8665-9666-4516-9895-3b023fb1aa8f)

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~

~~~

##### CLOUDSTACK VERSION

~~~

~~~

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~

~~~

##### ACTUAL RESULTS

~~~

~~~

**Comment by weizhouapache** [comment]
@DaanHoogland @sureshanaparti 
this seems to be critical issue

**Comment by weizhouapache** [comment]
rollback

```
update user_ip_address set account_id=1,domain_id=1,allocated=now(),state='Allocated' where public_ip_address='10.0.x.x';
```

**Comment by sureshanaparti** [comment]
> @DaanHoogland @sureshanaparti this seems to be critical issue

@weizhouapache Is this working in 4.18?

**Comment by weizhouapache** [comment]
> > @DaanHoogland @sureshanaparti this seems to be critical issue
> 
> @weizhouapache Is this working as expected, in 4.18?

@sureshanaparti 
unfortunately, 4.18 has the same issue

**Comment by shwstppr** [comment]
@weizhouapache @sureshanaparti @DaanHoogland if you're not already looking into this, I can have a look tomorrow

**Comment by DaanHoogland** [comment]
```
        // don't allow releasing system ip address
        if (ipVO.getSystem()) {
            throwInvalidIdException("Can't release system IP address with specified id", ipVO.getUuid(), "systemIpAddrId");
        }
```
in the `disassociateIpAddress` handler of `NetworkService`, `releaseIpAddress`.
And i checked the DB, something is going wrong in the aquiring of `user_ip_adresses` for system vms. No `isSystem` flag is set or even `account`, `allocated` time or even `state` (this one remains Free). I am still looking where to fix it but it seems to me that allocating an IP for a SVM should set the `is_system` flag on the record. I tested that by a DB hack and it indeed does the trick.

of course unsetting the flag needs to happen as well when releasing/destroying the SVM.

**Comment by DaanHoogland** [comment]
created #9281 for this but so far it only solves it for SSVM (and not yet tested)

**Comment by sureshanaparti** [comment]
Fixed in #9281
